src/app/components/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -9,19 +11,65 @@ import { Subscription } from 'rxjs';
 })
 export class BreadcrumbsComponent implements OnInit, OnDestroy {
   public isAuthenticated: boolean;
+  public breadcrumbs: IBreadCrumb[];
 
   private subscriptions = new Subscription();
 
-  constructor(private authService: AuthService) {}
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute
+  ) {}
 
   public ngOnInit(): void {
+
+    this.breadcrumbs = this.buildBreadCrumb(this.activatedRoute.root);
     this.isAuthenticated = this.authService.isAuthenticated();
     this.subscriptions.add(this.authService.isAuth$.subscribe((isAuth) => {
-      this.isAuthenticated = isAuth
+      this.isAuthenticated = isAuth;
+    }));
+
+    this.subscriptions.add(this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        this.breadcrumbs = this.buildBreadCrumb(this.activatedRoute.root);
+      }
     }));
   }
-  
+
   public ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
+
+  buildBreadCrumb(route: ActivatedRoute, url: string = '', breadcrumbs: IBreadCrumb[] = []): IBreadCrumb[] {
```

Comment:
Could you please add defenition type for this method 'public' or 'private'?

Reply:
Added private modificator